common-api/src/main/java/org/pmiops/workbench/cdr/dao/ConceptService.java

```diff
@@ -19,74 +19,102 @@
-      List<String> domainIds, int limit) {
-    Specification<Concept> conceptSpecification =
-        (root, criteriaQuery, criteriaBuilder) -> {
-            List<Predicate> predicates = new ArrayList<>();
-
-            // Check that the concept name, code, or ID matches the query string.
-            List<Predicate> queryPredicates = new ArrayList<>();
-            Expression<Double> matchExp = criteriaBuilder.function("match", Double.class,
-                root.get("conceptName"), criteriaBuilder.literal(query));
-            queryPredicates.add(criteriaBuilder.greaterThan(matchExp, 0.0));
-            queryPredicates.add(criteriaBuilder.equal(root.get("conceptCode"),
-                criteriaBuilder.literal(query)));
-            try {
-              long conceptId = Long.parseLong(query);
-              queryPredicates.add(criteriaBuilder.equal(root.get("conceptId"),
-                  criteriaBuilder.literal(conceptId)));
-            } catch (NumberFormatException e) {
-              // Not a long, don't try to match it to a concept ID.
-            }
-            predicates.add(criteriaBuilder.or(queryPredicates.toArray(new Predicate[0])));
-
-            // Optionally filter on standard concept, vocabulary ID, domain ID
-            if (standardConceptFilter.equals(StandardConceptFilter.STANDARD_CONCEPTS)) {
-                predicates.add(criteriaBuilder.equal(root.get("standardConcept"),
-                    criteriaBuilder.literal(STANDARD_CONCEPT_CODE)));
-            } else if (standardConceptFilter.equals(StandardConceptFilter.NON_STANDARD_CONCEPTS)) {
-              List<Predicate> standardConceptPredicates = new ArrayList<>();
-              standardConceptPredicates.add(criteriaBuilder.isNull(root.get("standardConcept")));
-              standardConceptPredicates.add(criteriaBuilder.notEqual(root.get("standardConcept"),
-                  criteriaBuilder.literal(STANDARD_CONCEPT_CODE)));
-              predicates.add(criteriaBuilder.or(
-                  standardConceptPredicates.toArray(new Predicate[0])));
-            }
-            if (vocabularyIds != null) {
-              predicates.add(root.get("vocabularyId").in(vocabularyIds));
+    public enum StandardConceptFilter {
+        ALL_CONCEPTS,
+        STANDARD_CONCEPTS,
+        NON_STANDARD_CONCEPTS
+    }
+
+    @PersistenceContext(unitName = "cdr")
+    private EntityManager entityManager;
+
+    public ConceptService() {
+    }
+
+    // Used for tests
+    public ConceptService(EntityManager entityManager) {
+        this.entityManager = entityManager;
+    }
+
+    public static String modifyMultipleMatchKeyword(String query){
+        // This function modifies the keyword to match all the words if multiple words are present(by adding + before each word to indicate match that matching each word is essential)
+        String[] keywords = query.split("[,+\\s+]");
+        for(int i = 0;i < keywords.length;i++){
```

Comment:
Please fix the spacing in this function. Should be a space after semicolons, before and after operators, etc.

Reply:
Fixed it.